web/files/files.go

```diff
@@ -307,13 +307,23 @@ func FileCopyHandler(c echo.Context) error {
 		return WrapVfsError(err)
 	}
 
+	copyName := c.QueryParam("Name")
+	if copyName == "" || filepath.Ext(copyName) != filepath.Ext(olddoc.DocName) {
```

Comment:
So, if a user wants to copy `FOO.JPG` to `bar.jpg`, it won't be accepted?

Reply:
Ah, I hadn't thought `filepath.Ext` could return different values for the same extension. Do you think this function is reliable enough to just strip the extension in the given name and use the original one?